### PR TITLE
maps: remove configmap

### PR DESCRIPTION
**Pull request by ianvernon**
`pkg/maps/configmap`, added in #6208, was used to add a means of looking up
information regarding endpoint configuration that could be plumbed before
compilation was performed, specifically that related to policy. The only use of
it in the datapath was for IPv4 ingress, to allow all based off of the state in
the configmap instead of checking the policymap for the given endpoint. This was
useful for when we had to do a compilation per-endpoint when Cilium was under
high load. But, now that we have the ELF templating + an allow-all for endpoints
being represented by a single PolicyMap entry, we do not have to worry about
such cases anymore. Remove the configmap accordingly and all references to it
from the datapath and from `pkg/endpoint`. Be sure to clean up old instances of
it after regeneration has succeeded for an endpoint, as well as in the stale map
cleanup function so that we do not leave old configmaps behind on the
filesystem.

Signed-off by: Ian Vernon <ian@cilium.io>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8895)

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
Will rebase to address SHA conflict after I get reviews to avoid unnecessarily running the CI.

File: pkg/endpoint/policy.go [hunk 1]
```diff
@@ -363,7 +363,9 @@ func (e *Endpoint) updateRealizedState(stats *regenerationStatistics, origDir st
 		e.syncPolicyMapController()
 	}
 
-	e.realizedBPFConfig = e.desiredBPFConfig
+	// Regeneration was successful - we can remove any state from prior versions
+	// of Cilium that needs to be cleaned up.
+	e.RemoveOldMaps()
```
Comment (joestringer) [inline pkg/endpoint/policy.go]: + "Remove after v1.7"

Good catch, I didn't think of this upgrade case.

**Comment by ianvernon** [comment]
test-me-please

File: pkg/endpoint/policy.go (hunk 1 above)
Comment (joestringer) [reply pkg/endpoint/policy.go]: Actually, why can't this be in the map sweeper? AFAICT that's only run after `<-restoreComplete` in the daemon so it should be safe to remove all `ep_config_*` at that point..?

**Review by joestringer — CHANGES_REQUESTED** [review]
I was under the impression that we still don't generate the "allow all" `{0,0}` datapath policy map entry for ingress, have you seen otherwise? Was expecting some change in policy / key generation to be necessary. But maybe I'm wrong on this front.

Minor nit below, seems like we might be able to contain the deprecated configmap handling to the map sweeper.

**Comment by ianvernon** [comment]
@joestringer I thought that the default value for allow-all on ingress (all zeroes) doesn't show up when dumping from the datapath, even though we do allow it? We have the logic in `DistillPolicies` and for `AllowAllIdentities`:

https://github.com/cilium/cilium/blob/82f6aba78b6f4d00e0083cac7ace4fea6f1849f2/pkg/policy/resolve.go#L106-L142

https://github.com/cilium/cilium/blob/82f6aba78b6f4d00e0083cac7ace4fea6f1849f2/pkg/policy/mapstate.go#L84-L106

File: pkg/endpoint/policy.go (hunk 1 above)
Comment (ianvernon) [reply pkg/endpoint/policy.go]: You're right, removed the need for the explicit function.

**Comment by ianvernon** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]
I don't think we have tests for "allow-all" policy..?

Either way, presumably the existing CI tests should fail if the changes in this PR broke ingress allow-all policy, as that would be necessary even in the "Policy enforcement disabled" case.

Looks good, would be nice to find out more about the missing entry when dumping policymap from the CLI.

**Review by tgraf — APPROVED** [review]

**Review by raybejjani — APPROVED** [review]

**Review by vadorovsky — APPROVED** [review]
Looks good. I will be happy to merge after rebase.

**Pull request merged**